etl/steps/data/garden/usgs/2024-07-15/mineral_commodity_summaries.py

```diff
@@ -0,0 +1,582 @@
+    if (year in [2023, 2024]) & (mineral == "GRAPHITE (NATURAL)"):
+        # Column "Reserves_t" contains rows that say 'Included in World total.' (and again, it has no notes).
+        # Make them nan and include them in notes.
+        index_issue = d[d["Reserves_t"] == "Included in World total."].index
+        d.loc[index_issue, "Reserves_t"] = pd.NA
+        assert "Reserves_notes" not in d.columns
+        d["Reserves_notes"] = pd.NA
+        d.loc[index_issue, "Reserves_notes"] = "Included in World total."
+    if (year in [2023, 2024]) & (mineral == "TITANIUM MINERAL CONCENTRATES"):
+        # One row contains "Included with ilmenite" (and 'included with ilmenite') in all data columns (but for
+        # whatever reason not in any of the "notes" columns).
+        # For simplicity, simply remove this row.
+        d = d[d["Reserves_kt"].str.lower() != "included with ilmenite"].reset_index(drop=True)
+
+    ############################################################################################################
+
+    # Check that all columns are either source, country, type, production, reserves, or capacity.
+    assert all(
+        [column.lower().startswith(("source", "country", "type", "prod_", "reserves_", "cap_")) for column in d.columns]
+    )
+    # Sanity checks.
+    assert d["Source"].unique().item() == f"MCS{year}"
+    assert "Country" in d.columns
+    assert "Type" in d.columns
+
+    # Sometimes there is a "Prod_notes" and sometimes a "Prod_note" column.
+    # For consistency, rename the latter to "Production_notes".
+    d = d.rename(columns={"Prod_note": "Production_notes", "Prod_notes": "Production_notes"}, errors="ignore")
+
+    # Add a column for the mineral name (although it will usually be very similar to "Type").
+    d["Mineral"] = mineral.capitalize()
+
+    return d
+
+
+def clean_spurious_values(series: pd.Series) -> pd.Series:
+    series = series.copy()
+    # Remove spurious spaces like "   NA".
+    series = series.fillna("").astype(str).str.strip()
+    for na_value in NA_VALUES:
+        series = series.replace(na_value, pd.NA)
+    # Remove spurious commas from numbers, like "7,200,000".
+    series = series.str.replace(",", "", regex=False)
+    # There is also at least one case (2023 Nickel) of ">100000000". Remove the ">".
+    series = series.str.replace(">", "", regex=False)
+    # There is also at least one case (2022 Helium) of "<1". Remove the "<".
+    series = series.str.replace("<", "", regex=False)
+    # There is also at least one case (2022 Vermiculite) of 'Less than ½ unit.'. Replace it by "0.5".
+    series = series.str.replace("Less than ½ unit.", "0.5", regex=False)
+    # There is also at least one case (2024 Sand and gravel (industrial)) of numbers starting with "e", e.g. "4200".
+    # I suppose that probably means "estimated" (and the corresponding notes mention the word "estimated").
+    # Remove those "e".
+    series = series.str.replace("e", "", regex=False)
```

Comment:
I'd suggest passing a dictionary to `replace`, can make your could more readable.

```suggestion
    series = series.str.replace(
        {
            # Remove spurious commas from numbers, like "7,200,000".
            ",": "",
            # There is also at least one case (2023 Nickel) of ">100000000". Remove the ">".
            ">": "",
            # There is also at least one case (2022 Helium) of "<1". Remove the "<".
            "<": "",
            # There is also at least one case (2022 Vermiculite) of 'Less than ½ unit.'. Replace it by "0.5".
            "Less than ½ unit.": "0.5",
            # There is also at least one case (2024 Sand and gravel (industrial)) of numbers starting with "e", e.g. "4200".
            # I suppose that probably means "estimated" (and the corresponding notes mention the word "estimated").
            # Remove those "e".
            "e": "",
        },
        regex=False
    )    
```

Reply:
This doesn't work, but I've changed it for better performance in the following way:
```
    series = (
        # Remove spurious commas from numbers, like "7,200,000".
        series.str.replace(",", "", regex=False)
        # There is also at least one case (2023 Nickel) of ">100000000". Remove the ">".
        .str.replace(">", "", regex=False)
        # There is also at least one case (2022 Helium) of "<1". Remove the "<".
        .str.replace("<", "", regex=False)
        # There is also at least one case (2022 Vermiculite) of 'Less than ½ unit.'. Replace it by "0.5".
        .str.replace("Less than ½ unit.", "0.5", regex=False)
        # There is also at least one case (2024 Sand and gravel (industrial)) of numbers starting with "e", e.g. "4200".
        # I suppose that probably means "estimated" (and the corresponding notes mention the word "estimated").
        # Remove those "e".
        .str.replace("e", "", regex=False)
    )
```